ipn/store.go

```diff
@@ -75,12 +75,14 @@ func CurrentProfileKey(userID string) StateKey {
 type StateStore interface {
 	// ReadState returns the bytes associated with ID. Returns (nil,
 	// ErrStateNotExist) if the ID doesn't have associated state.
+	// Implementations of ReadState are expected to be thread-safe.
```

Comment:
move these up to StateStore and:

```
//
// Implementations of StateStore are expected to be safe for concurrent use.
```

(Go generally doesn't say "thread safe", annoyingly)